### PR TITLE
8237372: NullPointerException in TabPaneSkin.stopDrag

### PR DESCRIPTION
**Pull request by effad**
Test simulates a single mouse-released event.
Fix simply guards against the null case.

## Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

## Issue
[JDK-8237372](https://bugs.openjdk.java.net/browse/JDK-8237372): NullPointerException in TabPaneSkin.stopDrag

## Approvers
 * Ambarish Rapte ([arapte](@arapte) - **Reviewer**)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
@arapte - can you review this?

The only question I have is whether is is expected that `dragTabHeader` can be null. If we understand why it is null, and that it isn't an error for it to be null, then the fix is fine. If its being null is unexpected, then the fix might just be masking the real problem. There isn't enough information in this PR to be able to evaluate this.

File: modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java [hunk 1]
```diff
@@ -2213,14 +2213,16 @@ private void stopDrag() {
...
         // Animate tab header being dragged to its final position.
-        dragHeaderSourceX = dragTabHeader.getLayoutX();
-        dragHeaderTransitionX = dragHeaderDestX - dragHeaderSourceX;
-        dragHeaderAnim.playFromStart();
-
-        // Reorder the tab list.
-        if (dragHeaderStartX != dragHeaderDestX) {
-            ((TabObservableList<Tab>) getSkinnable().getTabs()).reorder(dragTabHeader.tab, swapTab);
-            swapTab = null;
+        if (dragTabHeader != null) {
```
[the remark this answers is not in the document]
Comment (effad) [reply modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java]: You're right, this makes the method cleaner and easier to understand. I've adapted the PR.

File: modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java [hunk 2]
```diff
@@ -1954,6 +1954,7 @@ public Object queryAccessibleAttribute(AccessibleAttribute attribute, Object...
                 completeHeaderReordering();
             });
         }
+        @Override
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java]: This is unrelated to your change. Please revert it.

File: modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java [hunk 3]
```diff
@@ -1970,6 +1971,7 @@ protected void interpolate(double frac) {
                 resetDrag();
             });
         }
+        @Override
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java]: Please revert.

File: modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java [hunk 4]
```diff
@@ -1989,6 +1991,7 @@ private void removeReorderListeners(Node n) {
     }
 
     private ListChangeListener childListener = new ListChangeListener<Node>() {
+        @Override
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java]: Please revert.

Comment (effad) [reply modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java]: Reverted the @Overrides inserted by eclipse.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/TabPaneTest.java [hunk 5]
```diff
@@ -463,6 +464,24 @@ private void show() {
...
+
+        double xval = (tabPane.localToScene(tabPane.getLayoutBounds())).getMinX();
+        double yval = (tabPane.localToScene(tabPane.getLayoutBounds())).getMinY();
+
+        SceneHelper.processMouseEvent(scene,
+            MouseEventGenerator.generateMouseEvent(MouseEvent.MOUSE_RELEASED, xval+75, yval+20));
+        tk.firePulse();
+    }
+
+
```
Comment (arapte) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/TabPaneTest.java]: Please remove the additional empty line from here.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/TabPaneTest.java [hunk 6]
```diff
@@ -54,6 +54,7 @@
 import javafx.scene.Group;
 import javafx.scene.Scene;
 import javafx.scene.input.KeyEvent;
+import javafx.scene.input.MouseButton;
```
Comment (arapte) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/TabPaneTest.java]: This import it not used, please remove.

Comment (effad) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/TabPaneTest.java]: Removed import and extra empty line.

**Comment by kevinrushforth** [comment]
I don't have any additional comment, so this is good to go with 1 reviewer. @arapte can sponsor this.
